### PR TITLE
add a beforeSend callback

### LINKED ISSUE #34: implement beforeSend
jquery has this `beforeSend` callback that sets custom headers.

### PR DESCRIPTION
**Pull request by pYr0x**
close #34

### CODE REVIEW

File: can-ajax.js [hunk 1]
```diff
@@ -227,11 +228,15 @@ function ajax(o) {
 		xhr.setRequestHeader("X-Requested-With", "XMLHttpRequest");
 	}
 
-        if (o.xhrFields) {
-            for (var f in o.xhrFields) {
-                xhr[f] = o.xhrFields[f];
-            }
-        }
+	if(o.beforeSend){
+		o.beforeSend.call( null, xhr, o );
```
Comment (justinbmeyer) [inline can-ajax.js]: is `null` important here?  Might folks assume `this` works against the options (`o`)?

**Comment by justinbmeyer** [comment]
Thanks for the PR!

I think it's worth a tiny discussion on what should be in or out of `can-ajax`.  While this PR is fine as it adds so little code, what are our goals around jQuery compatibility?  For example, jQuery also supports many more complex options.

**Comment by pYr0x** [comment]
the same as #30, i want to add a JWT token on the request by using  the header `Authorization: Bearer 123`

**Comment by justinbmeyer** [comment]
@pYr0x any thoughts about `.call(null,`?

**Comment by pYr0x** [comment]
yeah i will change it to the context of `options`

**Pull request closed without merging**